### PR TITLE
[Bugfix][Quark] Fix W8A8 INT8 garbage outputs on Step-3.5-Flash (and other 3-key fused-MoE Quark exports)

### PR DESCRIPTION
**Pull request by JoursBleu**
### Summary

After quantizing `stepfun-ai/Step-3.5-Flash` to W8A8 INT8 with AMD Quark, serving the checkpoint via vLLM produces garbage tokens (random tokens, frequent NaN logits). If the same weights are offline-rewritten into compressed-tensors format and served by stock vLLM (with no changes whatsoever), everything works fine — this confirms that the INT8 numerics themselves are correct and the bug is in vLLM's Quark loading path. This PR fixes three independent bugs in the Quark INT8 backend; together they are the root cause of the garbage output. After the fix, the Quark-native checkpoint (`stepfun-ai/Step-3.5-Flash-Quark-W8A8-INT8`) runs directly without any rewrite, and matches the compressed-tensors baseline on GSM8K and is lossless against the BF16 baseline.

The change is scoped to the Quark INT8 path and the single model class `Step3p5ForCausalLM`; **the compressed-tensors path is untouched**.

### Three root causes

**1. `Step3p5ForCausalLM` is missing `packed_modules_mapping`.**

vLLM fuses `q_proj`/`k_proj`/`v_proj` into `qkv_proj` and `gate_proj`/`up_proj` into `gate_up_proj` (for both the dense MLP and the shared expert). `QuarkConfig.get_quant_method` calls:

```python
should_ignore_layer(prefix, ignore=exclude_layers,
                    fused_mapping=self.packed_modules_mapping)
```

When the model class does not declare `packed_modules_mapping` (the default empty dict), `should_ignore_layer` cannot reverse the fused prefix back to the unfused names listed in `quantization_config.exclude`.

In Quark exports, the shared expert is listed in unfused form in `exclude` (`model.layers.X.share_expert.{gate,up,down}_proj`). As a result, `share_expert.gate_up_proj` is not recognized as excluded and is wrongly attached as a Quark INT8 linear, even though the checkpoint contains no INT8 weights or scales for it — `weight_scale` stays at its zero-initialized value, the shared-expert output becomes identically 0 / NaN, and the corruption then propagates through residual + layernorm and ruins the entire forward pass.

After adding the standard `packed_modules_mapping` to `Step3p5ForCausalLM`, `should_ignore_layer` correctly matches the shared expert against the exclude list, and the layer falls back to `UnquantizedLinearMethod` and loads correctly.

**2. `QuarkW8A8MoEMethod` does not handle Quark's INT8 checkpoint layout.**

Quark's HF safetensors export differs from compressed-tensors only "cosmetically":

- It carries redundant symmetric `weight_zero_point` / `input_zero_point` tensors.
- Per-channel `weight_scale` is stored as 1D `[N]` per expert, with no trailing singleton dimension.

The old Quark MoE method could limp along with the legacy per-tensor INT8 recipe, but Step-3.5 (and similar models) use the modern per-channel + dynamic per-token activation recipe, and the old code path produced incorrect numerics on it.

This PR rewrites `QuarkW8A8Int8MoEMethod` to mirror `CompressedTensorsW8A8Int8MoEMethod` exactly:

- The per-channel scale is registered as 2D `[E, N]` for loading, then unsqueezed to `[E, N, 1]` in `process_weights_after_loading` so the modular int8 MoE kernel sees exactly the same layout as compressed-tensors.
- The per-channel path invokes the kernel via `int8_w8a8_moe_quant_config(..., per_act_token_quant=True)`.
- The symmetric `*_zero_point` tensors are registered as placeholders so safetensors loading does not fail on unknown keys, and are dropped in `process_weights_after_loading` before the quant config is built.
- The legacy per-tensor static W8A8 recipe is fully preserved (the shard-merging logic is unchanged), keeping backward compatibility.

**3. The non-MoE `QuarkW8A8Int8` linear also needs 2D scale alignment.**

For ordinary linears, Quark likewise stores per-channel `weight_scale` as 1D `[N]`, while the rest of vLLM and compressed-tensors expect 2D `[N, 1]`. This PR adds a small weight-loader shim that unsqueezes 1D scales before the standard shape check runs, and the parameter itself is allocated as `[N, 1]`. The runtime layout is then identical to compressed-tensors.

### Validation

`stepfun-ai/Step-3.5-Flash-Quark-W8A8-INT8` on MI355X, TP=2 + EP=2, vLLM `0.19.2rc1.dev109+gaad88f848.rocm721`:

|   | Smoke (`"What is 12 times 17?"`) | GSM8K (1319 q, 8-shot, max_tokens=8192) |
|---|---|---|
| before | garbage | — |
| **after** | "The product of 12 and 17 is **204**." | **1255 / 1319 = 95.15%** |
| Quark→CT-format baseline | OK | 1254 / 1319 = 95.07% |

BF16 baseline is 95.91%; the gap is within noise (lossless).

### Scope / risk

- Only `Step3p5ForCausalLM` and the Quark INT8 backend are touched; other models / other quantization paths are unaffected.
- The `packed_modules_mapping` added to `Step3p5ForCausalLM` is identical to what every other QKV+gate_up model in vLLM ships, and it only affects how Quark/CT `exclude` lists are matched against fused module prefixes — it does not affect weight loading itself.
- The non-MoE `QuarkW8A8Int8` linear change is backward-compatible: per-tensor (scalar) scales are unchanged, and old 1D `[N]` per-channel scales are reshaped to `[N, 1]` at load time.
- The MoE rewrite preserves the legacy per-tensor INT8 path; this PR fixes the per-channel + dynamic-input path.

### Related

- `stepfun-ai/Step-3.5-Flash-Quark-W8A8-INT8`: the public checkpoint that exposes this bug.

### CODE REVIEW

**Comment by JoursBleu** [comment]
@BowenBao gentle ping — follow-up to #36320 for Step-3.5-Flash (per-channel + per-token-dynamic INT8). Could you take a look? Thanks!

**Review by BowenBao — APPROVED** [review]
LGTM, thanks @JoursBleu!

Could you share the link to the Step-3.5-Flash quark int8 model? I see that it's mentioned as a public checkpoint in the description.

**Comment by JoursBleu** [comment]
@BowenBao  Step-3.5-Flash quark int8 model：https://huggingface.co/nameistoken/Step-3.5-Flash-Quark-W8A8-INT8

File: vllm/model_executor/layers/quantization/quark/schemes/quark_w8a8_int8.py [hunk 1]
```diff
@@ -67,17 +83,21 @@ def create_weights(
         layer.register_parameter("weight", weight)
 
         # WEIGHT SCALE
+        # NOTE: Mirror compressed-tensors layout: per-channel scale is 2D
```
Comment (tjtanaa) [inline vllm/model_executor/layers/quantization/quark/schemes/quark_w8a8_int8.py]: There are too many comments, let's remove this bulk of comments since line 50 has stated the reason.

**Comment by JoursBleu** [comment]
@tjtanaa  Simplified code comments.

**Review by tjtanaa — APPROVED** [review]
LGTM
